api/src/main/java/org/geysermc/geyser/api/util/Position.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019-2023 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.api.util;
+
+/**
+ * Represents a 3D position in the world.
+ */
+public record Position(float x, float y, float z) {
+
+    Position from(double x, double y, double z) {
+        return new Position((float) x, (float) y, (float) z);
+    }
+
+    Position from(int x, int y, int z) {
+        return new Position((float) x, (float) y, (float) z);
+    }
+
+    Position from(float x, float y, float z) {
+        return new Position(x, y, z);
+    }
```

Comment:
public right? and i would recommend `of` since we're not really converting from one "position type" to another

Reply:
valid, but i will wait on adjusting these unless we choose to not to expose cloudburst math stuff

Reply:
> unless we choose to not to expose cloudburst math stuff

we aren't yet though right?

are these methods currently unused?

Reply:
currently, they're unused; and they can be dropped entirely with cloudburst math. Was more of an example for methods we would have to create to re-create. E.g. adding/subtracting another position and whatnot wouldn't even be covered here - i prefer going with cloudburst math for this very reason of not re-inventing the wheel